server.ts

```diff
@@ -36,45 +37,57 @@ const Routes = require('./routes');
 //===----=---=-=--=--===--=-===----=---=-=--=--===--=-===----=---=-=--=--===--=-//
 
 
-function run() {
-    return new Promise((resolve, reject) => {
-        app.prepare()
-            .then(() => {
-                Setup.checkForAllEnvironmentVariables();
-                Tools.configure(server);
-                StaticFiles.include(server);
-                // FrontEndPageMap.include(server, app);
-                Security.configure(server);
-                Routes.include(server);
-
-                server.use((req, res, next) => {
-                    let domains = {
-                        'brshooter.com': true,
-                        'www.brshooter.com': true
-                    };
-                    const hostname = domains[req.hostname] ? 'www.brshooter.com' : req.hostname;
-                    if (req.headers['x-forwarded-proto'] === 'http') {
-                        res.redirect(301, `https://${hostname}${req.url}`);
-                        return;
-                    }
-                    next();
-                });
-                server.use('*', (req, res, next) => createAPIEventMiddleware(req, res, next));
-                server.get('*', (req, res) => {
-                    return handle(req, res);
-                });
-
-                server.listen(PORT, (err) => {
-                    if (err) {
-                        throw err;
-                    }
-
-                    resolve(server);
-
-                    logger.trace('Server Listening http://localhost:' + PORT);
-                });
-            }).catch((ex) => {
-            console.error(ex.stack);
+async function run() {
+    return new Promise(async (resolve, reject) => {
+        app.prepare().then(async () => {
+            Setup.checkForAllEnvironmentVariables();
+
+            //this checks for a valid database connection
+            try {
+                const db = await database;
```

Comment:
This checks for a valid database connection and throws a warning to the console on startup